pkg/skaffold/schema/versions.go

```diff
@@ -102,10 +100,6 @@ func ParseConfig(filename string, upgrade bool) (util.VersionedConfig, error) {
 		return nil, errors.Wrap(err, "unable to parse config")
 	}
 
-	if err := yamltags.ProcessStruct(cfg); err != nil {
```

Comment:
I put some explanation in the commit messages, but here I can be more verbose :). The fundamental problem with validation at this point is that after profile application, the configuration may become invalid. For example, if somebody configures a docker build in the base and applies a profile which _adds_ another kaniko build config. Profile application merges the two, so that the `oneOf=build` is violated, because both kaniko and docker are set in the result.
Therefore, I think it is better to generally not do validation here.